File: 2021/May/Day 17/findDuplicates.py
# Source :https://leetcode.com/problems/find-all-duplicates-in-an-array/submissions/
class Solution:
    def findDuplicates(self, nums: List[int]) -> List[int]:
        l =[]
        if(len(nums)==len(set(nums))):
            return l;
        nums=sorted(nums)

        prev = nums[0]
        for i in range(1,len(nums)):
            if(prev ==nums[i]):
                l.append(prev)
            prev = nums[i]
        return l

File: 2022/March/Day 22/FindDuplicates.cpp
// Source :https://leetcode.com/problems/find-all-duplicates-in-an-array/submissions/
class Solution {
public:
    vector<int> findDuplicates(vector<int>& nums) {
        sort(nums.begin(),nums.end());
        vector<int> out;

        int prev = nums[0];
        for(int i=1;i<size(nums);i++)
        {
            if(nums[i]==prev && std::find(out.begin(), out.end(), nums[i]) == out.end())
            {
                out.insert(out.end(),nums[i]);
            }
            prev=nums[i];

        }
        return out;
    }
};
